x-pack/plugins/maps/public/classes/sources/es_search_source/get_docvalue_source_fields.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IndexPattern } from '../../../../../../../src/plugins/data/common/index_patterns/index_patterns';
+import { getField } from '../../../../common/elasticsearch_util';
+
+export interface ScriptField {
+  source: string;
+  lang: string;
+}
+
+export function getDocValueAndSourceFields(
+  indexPattern: IndexPattern,
+  fieldNames: string[],
+  dateFormat: string
+): {
+  docValueFields: Array<string | { format: string; field: string }>;
+  sourceOnlyFields: string[];
+  scriptFields: Record<string, { script: ScriptField }>;
+} {
+  const docValueFields: Array<string | { format: string; field: string }> = [];
+  const sourceOnlyFields: string[] = [];
+  const scriptFields: Record<string, { script: ScriptField }> = {};
+  fieldNames.forEach((fieldName) => {
+    const field = getField(indexPattern, fieldName);
+    if (field.scripted) {
+      scriptFields[field.name] = {
+        script: {
+          source: field.script || '',
+          lang: field.lang || '',
+        },
+      };
+    }
+    // @ts-expect-error runtimeField has not been added to public API yet. exact shape of type TBD.
+    else if (field.readFromDocValues || field.runtimeField) {
```

Comment:
How about adding a functional test since the IFieldType API is in flux and we have disabled TS errors for the field. It would be a shame for the API to change and then this fix not work.

In esArchives, add the following, `"runtimeFieldMap" : """{"my geo.src":{"type":"keyword","script":{"source":"emit(doc[\"geo.src\"].value)"}}}""",`  to an index-pattern saved object and then test if field.runtimeField is working.